### PR TITLE
Use vendors when go generate

### PR DESCRIPTION
**Pull request by lunny**
Since we have store all the dependencies on vendor, I think we should also use them when run `go generate`. This will speed up `make generate` process and reduce failure possible when CI runs.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,7 +97,7 @@ vet:
 
 .PHONY: generate
 generate:
-	GO111MODULE=on $(GO) generate $(PACKAGES)
+	GO111MODULE=on $(GO) generate -mod=vendor $(PACKAGES)
```
Comment (techknowlogick) [inline Makefile:100]: if we add this, then we should make minimum supported version of go 1.11

Comment (lunny) [reply Makefile:100]: I think we have, since `make build` has already use `-mod=vendor` parameters.

Comment (techknowlogick) [reply Makefile:100]: Can you update here: https://github.com/go-gitea/gitea/blob/master/.drone.yml#L66 and https://github.com/go-gitea/gitea/blob/master/docs/content/doc/installation/from-source.en-us.md to say 1.11 is minimum

Comment (techknowlogick) [reply Makefile:100]: also on https://github.com/go-gitea/gitea/blob/master/docs/content/doc/advanced/hacking-on-gitea.en-us.md

**Comment by lunny** [comment]
@techknowlogick Maybe we should do that on v1.10, I will move this to v1.10 so that we could not change the minimal  Golang requirement on v1.9.

File: Makefile (hunk 1 above)
Comment (lunny) [reply Makefile:100]: done.

**Pull request merged**